### PR TITLE
Added docs directory to root

### PR DESCRIPTION
**Pull request by CaptainAMN**
Design doc and some other documentation

### CODE REVIEW

File: docs/rules-problems-solutions.txt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+be an OK idea for storage space, but still terrible for bandwidth available. So just a thought.
+
+
+
+    If we see anomalies, after all these measures, i think we can take further action. We will have to beta test this anyway.
+
+
+
+
+    Contact GitHub API Training Shop Blog About
```
Comment (nzyuzin) [inline docs/rules-problems-solutions.txt]: This line clearly looks off.

File: docs/rules-problems-solutions.txt [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
+
+Following added solutions to improve the matchmaking system. Shit ideas are enclosed by [.....]
```
Comment (nzyuzin) [inline docs/rules-problems-solutions.txt]: Please avoid obscenities and bad words.

File: docs/rules-problems-solutions.txt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ who is bottom scorer. He will not pick up where the leaver and topscorer A left. Rinse and repeat if this occasion happens multiple times.
+
+ This is quite experimental, needs beta testing.
+
+
+
+4) For capgames/Clans (not for TDM): If a league game is in progress, and someone leaves, first of all the game is fully frozen/ "paused". 
+   If a substitute is not found within 5 minutes countdown again, the game will be erased from database. ]
+
+---> Righfully according to Shiki, this is not possible with the current implementation.
```
Comment (nzyuzin) [inline docs/rules-problems-solutions.txt]: Style of that comment doesn't belong to documentation, please present the key idea for now, without additional info.

File: docs/rules-problems-solutions.txt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+[Now, for capgames, in some special circumstances, people can fuck around on purpose and make this a pain in the ass, if they decide just to
```
Comment (nzyuzin) [inline docs/rules-problems-solutions.txt]: Bad wording.

File: docs/rules-problems-solutions.txt [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+
+4) For capgames/Clans (not for TDM): If a league game is in progress, and someone leaves, first of all the game is fully frozen/ "paused". 
+   If a substitute is not found within 5 minutes countdown again, the game will be erased from database. ]
+
+---> Righfully according to Shiki, this is not possible with the current implementation.
+
+[Now, for capgames, in some special circumstances, people can fuck around on purpose and make this a pain in the ass, if they decide just to 
+leave and not come back. That's why we should keep the demo (if you can get it to work shiki) of these games, inspect the case by moderators, 
+and then take appropriate measures against the user. For clanwars, i see this as a minor problem at best. Leaving games should be simply punished,
```
Comment (nzyuzin) [inline docs/rules-problems-solutions.txt]: Personal opinions do not belong to documentation.

File: docs/rules-problems-solutions.txt [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  and other trouble like being afk while everyone warms up.
+
+    
+2) Leaving the game and not rejoining within 5 minutes earns you the same ranking as if you've lost the game and performed very poorly.
+
+
+
+3) A new player that comes and replaces someone gets the same points ingame as the worst player on the team.
+
+For example: If Sonix replaces Player A, the topscorer of team blue, Sonix will pick the game up with the same ingame points as Player D,
```
Comment (nzyuzin) [inline docs/rules-problems-solutions.txt]: Please avoid referring by names and use neutral 'Player'.

File: docs/Designdocument.txt [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+lack diversity. Secondly, to add a coefficient for undesired maps, so that your ranking does not get the "full" ranking boost. Else, a team of 3 players could single handedly push 
+the ranking into high points simply by only playing ctf3. Ergo, the points you potentially earn will be multiplied by a coefficient that is proprietary to the defined map.
+
+Alas, 
+
+ctf2 has the coefficient "1". 
+ctf3 has "0.5" ctf5 has "1.1" 
+ctf_tantum has "1".
+
+=This is an unfinished draft=
```
Comment (nzyuzin) [inline docs/Designdocument.txt]: Just leave it unfinished, no need to state it.

File: docs/Designdocument.txt [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+
+
++++++Special rules for clanwars:+++++
+
+Since all of this is based on ELO, we have to discourage repetitive maps and small teams. We have two options here. Seperate ELO rankings per map, what would be more accurate, but 
+lack diversity. Secondly, to add a coefficient for undesired maps, so that your ranking does not get the "full" ranking boost. Else, a team of 3 players could single handedly push 
+the ranking into high points simply by only playing ctf3. Ergo, the points you potentially earn will be multiplied by a coefficient that is proprietary to the defined map.
+
+Alas,
```
Comment (nzyuzin) [inline docs/Designdocument.txt]: Alas?

File: docs/Designdocument.txt [hunk 9]
```diff
@@ -0,0 +1,105 @@
+Design document is as follows:
```
Comment (nzyuzin) [inline docs/Designdocument.txt]: Redundant line.

File: docs/Designdocument.txt [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+
+
+
+**************Our goal:
```
Comment (nzyuzin) [inline docs/Designdocument.txt]: Here and in other places too many spaces between the lines.

File: docs/rules-problems-solutions.txt [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+   If a substitute is not found within 5 minutes countdown again, the game will be erased from database. ]
+
+---> Righfully according to Shiki, this is not possible with the current implementation.
+
+[Now, for capgames, in some special circumstances, people can fuck around on purpose and make this a pain in the ass, if they decide just to 
+leave and not come back. That's why we should keep the demo (if you can get it to work shiki) of these games, inspect the case by moderators, 
+and then take appropriate measures against the user. For clanwars, i see this as a minor problem at best. Leaving games should be simply punished, 
+like in every other online game.]
+
+---> We can never get this to work because of the required space to save all of the server side demos. NEEDS BETTER SOLUTION:
```
Comment (nzyuzin) [inline docs/rules-problems-solutions.txt]: Looks like redundant comment.
Can you style the solutions like pros-cons instead?

**Review by nzyuzin — CHANGES_REQUESTED** [review]

**Comment by CaptainAMN** [comment]
Hope this worked, thx for correcting.

**Comment by nzyuzin** [comment]
Thanks!

**Pull request merged**